### ISSUE TITLE
[4.x Design] Add blocking method to Scheduler

### ISSUE THREAD

**Issue by alexandru** [monix/monix#1377]
`Scheduler` should expose a `blocking` method:

```scala
trait Scheduler {
  // ...
  def blocking(r: Runnable): Unit
}
```

This is a complement to Scala's `BlockingContext`, except that it allows us to pick the right thread-pool on which to execute such blocking tasks.

Implementations could either reuse their main thread-pool, or setup a different thread-pool meant for blocking operations.

**Comment by Avasil** [comment]
> or setup a different thread-pool meant for blocking operations.

It would be a bit weird because `Scheduler` would become a pair of Schedulers. 
However, I think it would be nice UX when combined with `Task`. 
Similarily to CE3's `blocking` method. 
Currently, user needs to know to use `executeOn` and pick correct Scheduler.

**Comment by alexandru** [comment]
In fairness `Scheduler` is already a pair of thread-pools — because we need to support scheduling, coupled with thread-pool implementations that don't. `ForkJoinPool` doesn't do scheduling, but we add an internal `ScheduledExecutorService` that can take care of it.

Also `blocking` doesn't necessarily defer to another thread-pool. It could do nothing more but to use Scala's `BlockingContext`, or in case the `Scheduler` is simply a "cached thread pool", it could be just an alias of `execute`.

**Comment by alexandru** [comment]
I think that giving the ability to safely trigger blocking operations would make `Scheduler` much more valuable.

**Comment by jvican** [comment]
I agree. I think it gives the impression of supporting a pair of schedulers because we think in terms of execution thread pools. What really matters though is the overall abstraction. An scheduler can run different kinds of code intelligently depending on the context.
